src/language_server.ts

```diff
@@ -78,6 +83,85 @@ export class LanguageServer {
 	}
 
 
+	/**
+	 * Returns an array of string(s) representing the version of the Inmanta Language Server
+	 * that needs to be installed. The function checks if the environment variable INMANTA_LS_PATH is
+	 * set and uses it to install the LS from the specified path in editable mode. If the environment variable is not set,
+	 * it checks for the presence of requirements.txt file and installs the LS from it.
+	 * If requirements.txt file is not present, it installs the latest version of Inmanta Language Server.
+	 *
+	 * @returns {string[]} An array of string(s) representing the version of the Inmanta Language Server to be installed.
+	 */
+	languageServerVersionToInstall(): string[] {
+		const version = [];
+
+		if (process.env.INMANTA_LS_PATH) {
+		  version.push("-e", process.env.INMANTA_LS_PATH);
+		  log(`Installing Language Server from local source "${process.env.INMANTA_LS_PATH}"`);
+		} else {
+		  // Check for the presence of requirements.txt
+		  if (fs.existsSync(REQUIREMENTS_PATH)) {
+			version.push("-r", REQUIREMENTS_PATH);
+			log(`Installing Language Server from requirements file "${REQUIREMENTS_PATH}"`);
+		  } else {
+			version.push("inmantals");
+		  }
+		}
+
+		return version;
+	}
+
+	/**
+	 * Returns the version of the installed Inmanta Language Server, or null if it's not installed.
+	 *
+	 * @returns {string | null} The version of the installed Inmanta Language Server, or null if it's not installed.
+	 */
+	getInstalledInmantaLSVersion(): string | null {
+		try {
+		  const output = cp.execSync(`${this.pythonPath} -m pip show inmantals`).toString();
+		  const versionMatch = output.match(/^Version: (.+)$/m);
+		  if (versionMatch) {
+			return versionMatch[1];
+		  }
+		} catch (error) {}
+		return null;
+	}
+
+	/**
+	 * Checks if the correct version of the Inmanta Language Server is installed.
+	 *
+	 * @returns {boolean} True if the correct version of the Inmanta Language Server is installed, false otherwise.
+	 */
+	isCorrectInmantaLSVersionInstalled(): boolean {
+		// The LS is installed in editable mode via the env var
+		if (process.env.INMANTA_LS_PATH) {
+			return true;
+		}
+		// No requirements specified
+		if (!fs.existsSync(REQUIREMENTS_PATH)) {
+			return true;
+		}
+		// Get the expected version from requirement.txt
+		let expectedVersion = null;
+		const requirementTxtContent = fs.readFileSync(REQUIREMENTS_PATH, "utf-8");
+		const inmantaLSPattern = /^inmantals[=<>].*$/gm;
+		const inmantaLSLine = requirementTxtContent.match(inmantaLSPattern)?.[0];
+		if (inmantaLSLine) {
+			expectedVersion = inmantaLSLine.split(/[<=>]{2}/)[1];
```

Comment:
This only checks for the boundary version. This will suffice for `==` but not for other constraints. So either we should only support `==`, or we should make this a bit more robust. I think it would be nice if we could support `~=` if feasible (see my other comment), otherwise `==` will suffice. I would not make an effort to support the constraints we're not going to use. When we generate the file we can add a comment to it that mentions its purpose and that it only supports `~=`/`==`.

Reply:
By supporting `~=`, do you mean that it the user has version "1.2.3" of the LS, and that version "1.2.4" is released, he should also receive the pop-up to install the new version? because this I currently not support at all. I never check what is available on the pip index. Should I try that?

Reply:
That would be even better but I thought that might bring us too far. What I had in mind was that if the constraint is `~=1.2.3` and the user has `1.2.4` that we wouldn't keep asking them to upgrade.

Reply:
Ok i'll try to add support for what you propose. This should be feasible with not to much effort.